File: src/main/java/satisfiability/sample/Sample.java
package satisfiability.sample;

import satisfiability.formula.And;

public class Sample {
    public static void main(String[] args) {
        And and = new And();
        System.out.printf("true and false = %b%n", and.evaluate(true, false));
        System.out.printf("true and true = %b%n", and.evaluate(true, true));
    }
}

File: src/main/kotlin/satisfiability/sample/Sample.kt
package satisfiability.sample

import satisfiability.formula.And
import satisfiability.formula.StandardBooleanFormulaEvaluator


fun main() {

    val evaluator = StandardBooleanFormulaEvaluator()
    println("true and false = ${evaluator.eval(And(listOf("x1", "x2")), mapOf("x1" to true, "x2" to false))}")
    println("true and true = ${evaluator.eval(And(listOf("x1", "x2")), mapOf("x1" to true, "x2" to true))}")
}
